### DISCUSSION TITLE
ldap conection

### DISCUSSION THREAD

**Discussion by Froc88** [rejetto/hfs#181] [Ideas]
Hello!
Will there be any possibility to enable ldap to import users?

**Comment by rejetto** [comment]
i'm interested so "maybe". I'm not sure what the effort would be.

It's my understanding that ldap can be used in different ways/strategies. I don't know much.
Can you say more about your case?

btw, 0.42 is able to import from csv

**Reply by Froc88** [reply]
Currently I use your application to share files in an environment with 140 users managed by active directory.
The query was to make it easier for users to use the same password for both.

**Reply by rejetto** [reply]
do you have keycloak, perhaps?

**Reply by Froc88** [reply]
No, the other service I use is GLPI with ldap authentication.
keycloak nothing

**Comment by rejetto** [comment]
i guess ldap means we should have at least 2 different features
1) login, because i expect ldap won't let us copy in hfs the actual password, ldap server will need to be asked for each login attempt
2) synchronization of accounts, because if you don't get on HFS the list of users and groups, you won't be able to use them for permissions

thoughts, anyone?

**Reply by Froc88** [reply]
Yes, it would be to export the users with an ldap command using a user from the active directory.
Then validate the passwords against the AD.

**Comment by rejetto** [comment]
LDAP is now available as plugin for version 0.57 (still beta at this moment)

**Comment by Froc88** [comment]
Thanks!
I tried it and the connection was established, but how do I get it to import the contacts?
![image](https://github.com/user-attachments/assets/6d4a67f5-1046-421b-902a-ef340701ff6d)

**Comment by rejetto** [comment]
it is already importing contacts, but it says zero.
You probably have something to fix in the parameters you configured in the plugin.

**Reply by Froc88** [reply]
I have this setup
![image](https://github.com/user-attachments/assets/2af169eb-8ebf-468a-9a09-3878439fbd59)

**Comment by rejetto** [comment]
i'm no ldap expert, but can't see anything necessarily wrong in your config, but also can't verify if that config works for your server.
My advice is to use some command line tool to check if it is supposed to work or not, like ldapsearch: 
> ldapsearch -x -D "cn=ldap,cn=users,dc=dcr,dc=gov,dc=ar" -w "your_password_here" -b "dc=dcr,dc=gov,dc=ar" "(objectClass=user)" cn memberOf

ldap configuration can vary greatly

**Comment by Froc88** [comment]
If I run this it returns the users
ldapsearch -x -D "cn=ldap,cn=users,dc=dcr,dc=gov,dc=ar" -w "xxx" -b "dc=dcr,dc=gov,dc=ar" "(objectClass=user)" -H ldap://10.2.1.100 cn memberOf sale

**Comment by rejetto** [comment]
check if you have the "dn" field, as the plugin will ignore any account without it

**Comment by rejetto** [comment]
you set scope=base
could that be the problem?
apparently ldapsearch default to scope=sub
which is the same default of the plugin too, but you must have changed it

**Reply by Froc88** [reply]
If I choose scope base it returns records updated:0, scope one returns ReferenceError: a is not defined, scope sub returns ReferenceError: a is not defined

**Reply by rejetto** [reply]
well, that sounds like a bug

**Comment by rejetto** [comment]
can you tell me what version of the plugin do you have?

**Reply by Froc88** [reply]
0.24

**Comment by rejetto** [comment]
that's an old version.
when did you install it?
current version is 0.32

**Reply by Froc88** [reply]
I hadn't seen the update. I'll check to see if anything changes.
Thanks

**Comment by rejetto** [comment]
ok but can you tell me if you installed this plugin today for the first time?

**Comment by Froc88** [comment]
I had downloaded it previously.
I was able to sync my domain contacts, but when I try to authenticate with one of them, I get an error.
![image](https://github.com/user-attachments/assets/13db0149-f94e-43e8-925b-a1d8b26f84f8)

**Comment by rejetto** [comment]
this "error" is not helpful,
see if you can find any detail in the console of hfs.

**Comment by rejetto** [comment]
this screenshot, which is from the log, shows a request that did a normal authentication (and failing), not ldap.
are you sure it's an account from ldap?
Is this the only request for the failed login?
Do accounts that are not from ldap work?

**Comment by rejetto** [comment]
if you select the ldap account in Admin-panel > Accounts
do you see the "Authentication handled by plugin" ? 
<img width="445" alt="image" src="https://github.com/user-attachments/assets/788d4a39-122f-487d-b8e0-181df2f88e5f" />

**Comment by Froc88** [comment]
I found the problem. I had imported the display name and needed the account name.
I changed it and it worked perfectly.
The configuration was like this
![image](https://github.com/user-attachments/assets/897e5b08-4f35-4553-8a0f-cfe17e9ccf68)

**Comment by rejetto** [comment]
that's good!
thanks for sharing your findings.
After all, if you had left the "login field" at its default state, you wouldn't have had the problem :)
<img width="139" alt="image" src="https://github.com/user-attachments/assets/a1b876f3-1e9d-4532-a378-9050631e213d" />

**Comment by Froc88** [comment]
I'm still trying to figure out how to import users from the 3 OUs I have in the AD, since if I change the OU in the dn users base field, it deletes what I had added and imports the users from that OU.

**Comment by rejetto** [comment]
Are you sure you cannot do all of them with a single query?
If you can't, then I need to make it possible to specify multiple "Base DN users", i guess.

don't you have an OU that's including all the 3 OUs you want to import?

**Reply by Froc88** [reply]
Remove the ou and choose sub in scope.
![image](https://github.com/user-attachments/assets/c04957dc-2da5-4b5c-a779-136d1d3973c0)

**Comment by rejetto** [comment]
At the moment the plugin set a limit of 1000 accounts,
and i think this error means you would get more than that.
Do you want import more than 1000 ?

**Reply by Froc88** [reply]
275

**Comment by rejetto** [comment]
ok, you want 275, but that errors i think it means that the query with those parameter generated over 1000 accounts.
so, it's good that it stopped the process.
only thing is that the message may be clearer. I can try to do that, but that's a different problem.
So, i guess this configuration you tried is no good for what you want to do.
If you confirm, this means that you need to specify 3 OUs, and the plugin need to be able to do that.

**Reply by Froc88** [reply]
I was also importing the computers. I changed the user filter to (&(objectClass=user)(!(objectCategory=computer)))
It worked perfectly.
Thanks for your attention!

**Reply by rejetto** [reply]
ok, that's cool
